### ISSUE TITLE
list_connections says it takes a ConnectionType, takes a str

### ISSUE THREAD

**Issue by tyhoff** [workos-inc/workos-python#110]
Hello! 

The docs state that this function takes a ConnectionType, so we assumed it did, but it actually takes in a str. I've included the error below that we were seeing on our application.

It would be great to have so error handling as you do here: https://github.com/workos-inc/workos-python/blob/8dae7b1dc8c31db179e7c0e012faeb1d7fd55179/workos/sso.py#L84-L87

```
>>> workos.client.sso.list_connections(domain="foo-corp.com", connection_type=ConnectionType.OktaSAML)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/.heroku/python/lib/python3.8/site-packages/workos/sso.py", line 202, in list_connections
    return self.request_helper.request(
  File "/app/.heroku/python/lib/python3.8/site-packages/workos/utils/request.py", line 89, in request
    raise BadRequestException(
workos.exceptions.BadRequestException: (message=Validation failed, request_id=...)

(message=Validation failed, request_id=45831dcd-650a-4ac2-bd40-eb96717420f7)
```

```
{
  "date": "2022-02-16T00:12:50.856Z",
  "id": "...",
  "method": "GET",
  "path": "/connections",
  "status": 422,
  "source": "WorkOS Python/1.7.0",
  "request_query": "{\"connection_type\":\"ConnectionType.OktaSAML\",\"domain\":\"foo-corp.com\",\"limit\":\"10\"}",
  "request_body": null,
  "response_body": {
    "errors": [
      {
        "code": "connection_type must be a valid enum value",
        "field": "connection_type"
      }
    ],
    "message": "Validation failed"
  }
}
```

Running the 1.7.0 version on Python 3.8.

**Comment by blairlunceford** [comment]
Hi @tyhoff thanks for reporting this! We're looking into it 👍

**Comment by maxdeviant** [comment]
@tyhoff This has been addressed in v1.8.0 of the SDK.

The `connection_type` parameter for `list_connections` now accepts a `ConnectionType`.

String values are still accepted for backwards compatibility, but are considered deprecated.
